core/src/main/scala/org/apache/spark/status/api/v1/ApiRootResource.scala

```diff
@@ -222,6 +222,7 @@ private[spark] object ApiRootResource {
 private[spark] trait UIRoot {
   def getSparkUI(appKey: String): Option[SparkUI]
   def getApplicationInfoList: Iterator[ApplicationInfo]
+  def getApplicationInfoList(limit: Int): Iterator[ApplicationInfo]
```

Comment:
Based on the limited uses of `getApplicationInfoList` it would make more sense to me to just add the `limit` param it with a default value (`limit=-1` would return the whole list)